### DISCUSSION TITLE
Will development continue after official Google deprecation of Manifest v2 API, for other Chromium-based browsers will keep to support MV2 API?

### DISCUSSION THREAD

**Discussion by Half-Modern** [uBlockOrigin/uBlock-issues#3386] [Q&A] [answered]
Manifest V2 is being deprecated this year. There is a group policy to extend it until 2025, but after than that, Google Chrome (and most Chromium-based browsers) will disable any MV2-based extension and won't let them to be installed. They will also be removed from the Web Store.
Now the question is:
There are other browsers where their authors claim they will support MV2 even after Google's deprecation.
Will uBlock Origin (Original, Manifest v2) be maintained and updated for those browsers?
I hope the answer is yes, because I wouldn't imagine browsing the internet without it and Firefox just doesn't work good for me. Both matter of preferences and heavy usage of Google websites (mainly Youtube).

**Comment by gorhill** [comment]
See <https://github.com/uBlockOrigin/uBlock-issues/wiki/About-Google-Chrome's-%22This-extension-may-soon-no-longer-be-supported%22>

Other than what is written there, I don't make pledge for future commitments when I don't know what the future holds.

> Firefox just doesn't work good for me

Ultimately you make choices and consequently you have to accept the burden of the choices you make.

**Reply by Half-Modern** [reply, marked as answer]
> Ultimately you make choices and consequently you have to accept the burden of the choices you make.

I take this as no, so I'll gladly wait for an unofficial Fork for Supermium/Thorium rather than using a browser that performs worse just for ad-blocking.

**Comment by garry-ut99** [comment]
> FK2FAGHMS

Duplicate of:
- https://github.com/uBlockOrigin/uBlock-issues/discussions/1792
- https://github.com/uBlockOrigin/uBlock-issues/discussions/2987
- https://github.com/uBlockOrigin/uBlock-issues/discussions/3334

There is no need to create another threads, can still ask in the previous ones.

**Reply by Half-Modern** [reply]
> There is no need to create another threads, can still ask in the previous ones.

No developer response until I open a new thread, so looks like I NEEDED to open it.
Also; it's not a duplicate:
1st: asks if UBO Legacy really ends when MV3 is enforced. UBO is also available for Firefox, so not relevant to my question.
2nd:asks if it can support MV2 version until Chrome's extended MV2 support policy expires. Not related to my question either.
3rd: While it's kinda relevant to my question, it doesn't answer "Will developers continue working on uBO Legacy for unofficial Chromium Forks?", not a duplicate again.
Next time think twice before you accuse someone for opening a duplicate thread.

**Reply by garry-ut99** [reply]
> FK2FAGHMS : No developer response until I open a new thread, so looks like I NEEDED to open it.

Main developer not always must reply, he might be busy and have no time, or he might be not interested, or he might be not sure what will happen in the future, or he might considered the other replies as sufficient or any other reason. It should be respected, and one shouldn't  pressure dev to reply by creating new threads and exploit it as a way to draw attention. Also as far I know the main developer and other devs see notifications of new comments even in closed issues/discussions, hence a new comment in an old thread doesn't differ than a new thread, devs still will see it.

> FK2FAGHMS : Also; it's not a duplicate Next time think twice before you accuse someone for opening a duplicate thread. (...)

Very similiar questions still fall into category of duplicates, regardless of whether it's 100% duplicate or 80-90% similiarity, here is the example reasoning: `https://github.com/uBlockOrigin/uBlock-issues/issues/3358#issuecomment-2320225078` and I see it applies for the current case as well, especially given 3 other threads already exist.

> FK2FAGHMS : 3rd: While it's kinda relevant to my question, it doesn't answer "Will developers continue working on uBO Legacy for unofficial Chromium Forks?", not a duplicate again.

Seems 99% identical, and the only diff is currently vs future, you could ask your question in the previous thread, you're asking about the future: no one knows what will happen in the future.

> FK2FAGHMS :  I take this as no

He didn't say yes or no, just that noone can predict the future. And in the linked Wiki it says: "Will development of uBO continue? Yes, there are other browsers which are not deprecating Manifest v2, e.g. [Firefox](https://blog.mozilla.org/addons/2024/03/13/manifest-v3-manifest-v2-march-2024-update/)." The "e.g." stands for "for example" which means MV2 support will not be limited to Firefox, and Firefox is just an example. Also the "e.g. [Firefox](https://blog.mozilla.org/addons/2024/03/13/manifest-v3-manifest-v2-march-2024-update/)." " part was added later by a volunteer, and not by uBO creator, see [Wiki commit history ](https://github.com/uBlockOrigin/uBlock-issues/wiki/About-Google-Chrome's-%22This-extension-may-soon-no-longer-be-supported%22/_history) --> [commit by M3DZIK](https://github.com/uBlockOrigin/uBlock-issues/wiki/About-Google-Chrome's-%22This-extension-may-soon-no-longer-be-supported%22/_compare/ccc0c74dfe849212a9ed2195402eed184107c5ce...194f50030cedbb7e1151ad618aba059ee8be3e66). I've expanded it further: [commit](https://github.com/uBlockOrigin/uBlock-issues/wiki/About-Google-Chrome's-%22This-extension-may-soon-no-longer-be-supported%22/_compare/5923d6498b281a827208caeeec0614bb0128c088...7beaa9ee0be53fd3f026042d79878e078e978f02).

> FK2FAGHMS :  before you accuse someone

Using your words I could say that it's you who is accusing me of accusing you...or maybe noone is accusing noone, and everyone just said their opinions...Next time think twice before you accuse someone for accusing. There is no need go straight to accusation of accusing, this just poisons immediately the discussion. Just pointing out this is possibly not a duplicate in your opinion, is enough, ultimately devs will decide whether to close it as duplicate or not, and as seen they already labeled it as a "duplicate". Me too consider that 99% identical questions don't require a new separate thread and are duplicates.
